Fix UITextSearching protocol functions in site isolation

colelao opened the pull request:
#### 8c52c9461526901f003fb6f8482c42d7f8cabca9
<pre>
Fix UITextSearching protocol functions in site isolation
<a href="https://bugs.webkit.org/show_bug.cgi?id=302986">https://bugs.webkit.org/show_bug.cgi?id=302986</a>
<a href="https://rdar.apple.com/165243794">rdar://165243794</a>

Reviewed by Sihui Liu and Alex Christensen.

This patch fixes issues with the `UITextSearching` protocol
in site isolation.

It fixes `replaceFoundTextInRange`, `decorateFoundTextRange`,
`scrollRangeToVisible`, `clearAllDecoratedFoundText`, and
`requestRectForFoundTextRange` in site isolation.

The main problem is that a client can search for string in
one WebView and then highlight or replace or scroll in another
that is hosting the same content. We address this issue by storing
the path from the mainFrame to the frame in which the match was
found on each WebFoundTextRange.

This way, we can traverse the frame tree for each
WebFoundTextRange and find, positionally, which frame the range
was found in. This means we do not have to store the frameIdentifier,
which will be different between WebViews.

* Source/WebCore/page/Frame.cpp:
(WebCore::Frame::indexInFrameTreeSiblings const):
(WebCore::Frame::pathToFrame const):
* Source/WebCore/page/Frame.h:
* Source/WebCore/page/Page.cpp:
(WebCore::Page::findFrameByPath const):
* Source/WebCore/page/Page.h:
* Source/WebKit/Shared/WebFoundTextRange.cpp:
(WebKit::WebFoundTextRange::operator== const):
(WebKit::operator&lt;&lt;):
* Source/WebKit/Shared/WebFoundTextRange.h:
(WTF::HashTraits&lt;WebKit::WebFoundTextRange&gt;::constructDeletedValue):
(WTF::HashTraits&lt;WebKit::WebFoundTextRange&gt;::isDeletedValue):
* Source/WebKit/Shared/WebFoundTextRange.serialization.in:
* Source/WebKit/UIProcess/WebFrameProxy.cpp:
(WebKit::WebFrameProxy::childFrame const):
* Source/WebKit/UIProcess/WebFrameProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::replaceFoundTextRangeWithString):
(WebKit::WebPageProxy::scrollTextRangeToVisible):
(WebKit::WebPageProxy::clearAllDecoratedFoundText):
(WebKit::WebPageProxy::didBeginTextSearchOperation):
(WebKit::WebPageProxy::requestRectForFoundTextRange):
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(+[WKFoundTextRange foundTextRangeWithWebFoundTextRange:]):
(-[WKFoundTextRange dealloc]):
(-[WKFoundDOMTextRange webFoundTextRange]):
(-[WKFoundPDFTextRange webFoundTextRange]):
* Source/WebKit/WebProcess/WebPage/WebFoundTextRangeController.cpp:
(WebKit::WebFoundTextRangeController::findTextRangesForStringMatches):
(WebKit::WebFoundTextRangeController::rectsForTextMatchesInRect):
(WebKit::WebFoundTextRangeController::frameForFoundTextRange const):
* Tools/TestRunnerShared/spi/UIKitSPIForTesting.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPage.mm:
(TEST(WebKit, FindAndReplace)):
(swizzledIsEmbeddedScreen): Deleted.
(-[TestScrollViewDelegate init]): Deleted.
(-[TestScrollViewDelegate scrollViewDidEndScrollingAnimation:]): Deleted.
(-[TestFindDelegate setDidAddLayerForFindOverlayHandler:]): Deleted.
(-[TestFindDelegate didAddLayerForFindOverlayHandler]): Deleted.
(-[TestFindDelegate setDidRemoveLayerForFindOverlayHandler:]): Deleted.
(-[TestFindDelegate didRemoveLayerForFindOverlayHandler]): Deleted.
(-[TestFindDelegate _webView:didAddLayerForFindOverlay:]): Deleted.
(-[TestFindDelegate _webViewDidRemoveLayerForFindOverlay:]): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPageUtilities.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPageUtilities.mm:
(swizzledIsEmbeddedScreen):
(-[TestScrollViewDelegate init]):
(-[TestScrollViewDelegate scrollViewDidEndScrollingAnimation:]):
(-[TestFindDelegate setDidAddLayerForFindOverlayHandler:]):
(-[TestFindDelegate didAddLayerForFindOverlayHandler]):
(-[TestFindDelegate setDidRemoveLayerForFindOverlayHandler:]):
(-[TestFindDelegate didRemoveLayerForFindOverlayHandler]):
(-[TestFindDelegate _webView:didAddLayerForFindOverlay:]):
(-[TestFindDelegate _webViewDidRemoveLayerForFindOverlay:]):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm:
(TestWebKitAPI::(SiteIsolation, FindStringInFrameAndReplaceIOS)):
(TestWebKitAPI::(SiteIsolation, DecorateFoundTextRangeIOS)):
(TestWebKitAPI::(SiteIsolation, ScrollTextRangeToVisibleIOS)):
(TestWebKitAPI::(SiteIsolation, ClearAllDecoratedFoundTextIOS)):
(TestWebKitAPI::(SiteIsolation, RequestRectForFoundTextRangeIOS)):

Canonical link: <a href="https://commits.webkit.org/303851@main">https://commits.webkit.org/303851@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0671de4912f3395b6ca25ae7300b11915090917f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/133793 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/6303 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/44980 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/141368 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/85852 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/edfae664-3e32-4c13-89bb-2c6796cdca3b) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/135663 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/155/builds/6830 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/6166 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/102345 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/85852 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/7104fff8-df81-4f68-a15d-829a625dafd1) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/136740 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/155/builds/6830 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/119943 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/83146 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/e433dee3-1700-4204-bd0e-d4df5f74f6c4) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/155/builds/6830 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/2312 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [⏳ 🛠 wpe-cairo ](https://ews-build.webkit.org/#/builders/WPE-Cairo-Build-EWS "Waiting in queue, processing has not started yet") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/155/builds/6830 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/38061 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/144016 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/5972 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/38641 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/110720 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/160/builds/6056 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/161/builds/5102 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/110910 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28124 "Built successfully and passed tests") | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-26-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/116198 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/59721 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/6025 "Built successfully") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/34486 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/157/builds/5871 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/69489 "Built successfully") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/163/builds/6117 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/152/builds/5979 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPageUtilities.mm:
```diff
@@ -24,10 +24,18 @@
  */
 
 #import "config.h"
+#include <UIKitCore/UIKitCore.h>
```
We should not be importing UIKitCore.
```suggestion
#include <UIKit/UIKit.h>
```

pxlcoder replied:
You may also need a platform guard.

pxlcoder commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPageUtilities.mm:
```diff
@@ -24,10 +24,18 @@
  */
 
 #import "config.h"
+#include <UIKitCore/UIKitCore.h>
```
Also this is imported in the header too. Can we avoid that with forward declarations?

pxlcoder commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPageUtilities.mm:
```diff
@@ -24,10 +24,18 @@
+{
+    return NO;
+}
+
```
```suggestion
```

pxlcoder commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPageUtilities.mm:
```diff
@@ -109,4 +185,28 @@ void testPerformTextSearchWithQueryStringInWebView(WKWebView *webView, NSString
+
+void testRequestRectForFoundTextRange(WKWebView *webView, UITextRange *range, void(^completionHandler)(CGRect)) {
+    [webView _requestRectForFoundTextRange:range completionHandler:completionHandler];
+}
```
Why do we need these wrapper methods?

pxlcoder commented on Source/WebKit/WebProcess/WebPage/WebFoundTextRangeController.cpp:
```diff
@@ -527,14 +528,26 @@ Vector<WebCore::FloatRect> WebFoundTextRangeController::rectsForTextMatchesInRec
+    if (webFrame)
+        return webFrame->coreLocalFrame();
+
+    // HACK: some clients (mail) want to search in one WebView and highlight in another that hosts the same content
```
We don't need to say "HACK" 😅

pxlcoder commented on Source/WebKit/WebProcess/WebPage/WebFoundTextRangeController.cpp:
```diff
@@ -527,14 +528,26 @@ Vector<WebCore::FloatRect> WebFoundTextRangeController::rectsForTextMatchesInRec
+
+    RefPtr webFrame = WebFrame::webFrame(range.frameIdentifier);
+    if (webFrame)
+        return webFrame->coreLocalFrame();
```
I don't think this is ever right, given the below. If the same value ever occurs across two web views, and the order is different, we'd get the incorrect frame.

pxlcoder commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -6365,32 +6365,37 @@ void WebPageProxy::findTextRangesForStringMatches(const String& string, OptionSe
 void WebPageProxy::decorateTextRangeWithStyle(const WebFoundTextRange& range, FindDecorationStyle style)
 {
-    send(Messages::WebPage::DecorateTextRangeWithStyle(range, style));
+    sendToProcessContainingFrame(range.frameIdentifier, Messages::WebPage::DecorateTextRangeWithStyle(range, style));
```
How would this target the right frame / be sent to the right process if the result comes from a different web view?

pxlcoder replied:
Don't we need to do the order -> frame identifier mapping on the UI process side?

pxlcoder commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPageUtilities.h:
```diff
@@ -39,7 +43,24 @@
+@interface TestScrollViewDelegate : NSObject<UIScrollViewDelegate>  {
+    @public bool _finishedScrolling;
+
+    std::unique_ptr<InstanceMethodSwizzler> _isEmbeddedScreenSwizzler;
```
This does not need to be exposed in the header.

pxlcoder commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -6365,32 +6365,37 @@ void WebPageProxy::findTextRangesForStringMatches(const String& string, OptionSe
+    forEachWebContentProcess([&](auto& process, auto pageID) {
+        process.send(Messages::WebPage::ClearAllDecoratedFoundText(), pageID);
+    });
+
```
```suggestion
```

szewai commented on Source/WebKit/WebProcess/WebPage/WebFoundTextRangeController.cpp:
```diff
@@ -527,14 +528,26 @@ Vector<WebCore::FloatRect> WebFoundTextRangeController::rectsForTextMatchesInRec
+        currentOrder++;
+        if (currentOrder == range.order)
+            return dynamicDowncast<WebCore::LocalFrame>(frame.get());
+    }
```
If the WebKit client can get the range from one view's web process and apply it on another view, won't this change break the case where two views have frames with the same name but different frameIDs?

szewai commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -6365,7 +6365,7 @@ void WebPageProxy::findTextRangesForStringMatches(const String& string, OptionSe
 void WebPageProxy::replaceFoundTextRangeWithString(const WebFoundTextRange& range, const String& string)
 {
-    send(Messages::WebPage::ReplaceFoundTextRangeWithString(range, string));
+    sendToProcessContainingFrame(range.frameIdentifier, Messages::WebPage::ReplaceFoundTextRangeWithString(range, string));
```
Don't we need to `findFrameByName` and use the identifier of the found frame here, as the range can be generated from a different view (that has different frame identifier)?

colelao replied:
The only use case I know of where two WebViews can be used is with search and highlight (`requestRectForFoundTextRange`), but it might be a good idea to do this for the other functions as well from a defensive standpoint. What do you think?

szewai replied:
I think it might be safer to separate the behavior change and site isolation support, i.e. adding the site isolation support without relying on frame identifier in this PR and adding frame identifier in a followup PR; so that if the behavior change breaks anything we can just revert the latter commit.

szewai commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -6375,22 +6375,52 @@ void WebPageProxy::decorateTextRangeWithStyle(const WebFoundTextRange& range, Fi
 void WebPageProxy::scrollTextRangeToVisible(const WebFoundTextRange& range)
 {
-    send(Messages::WebPage::ScrollTextRangeToVisible(range));
+    sendToProcessContainingFrame(range.frameIdentifier, Messages::WebPage::ScrollTextRangeToVisible(range));
```
Don't we need to `findFrameByName` and use the identifier of the found frame here?

achristensen07 reviewed (changes requested):
I don't think we should be using uniqueName for anything more than legacy layout test results.  Can't FrameIdentifier uniquely identify a frame?

achristensen07 commented on Source/WebCore/page/Frame.cpp:
```diff
@@ -209,6 +210,35 @@ Ref<NavigationScheduler> Frame::protectedNavigationScheduler() const
+    }
+
+    ASSERT_NOT_REACHED("This frame should be in its own tree");
+    return 0;
```
This should indeed never happen, but I think it would be slightly better to return nullopt here.

achristensen07 commented on Source/WebCore/page/Frame.cpp:
```diff
@@ -209,6 +210,35 @@ Ref<NavigationScheduler> Frame::protectedNavigationScheduler() const
+Vector<size_t> Frame::pathToFrame() const
+{
+    Vector<size_t> path;
+    auto current = this;
```
RefPtr instead of auto would probably be a bit "safer"

achristensen07 commented on Source/WebCore/page/Frame.cpp:
```diff
@@ -209,6 +210,35 @@ Ref<NavigationScheduler> Frame::protectedNavigationScheduler() const
+    auto current = this;
+
+    while (current) {
+        if (auto index = current->indexInFrameTreeSiblings(); index.has_value())
```
"; index.has_value()" can be omitted without changing behavior.

achristensen07 commented on Source/WebCore/page/FrameTree.cpp:
```diff
@@ -346,6 +346,22 @@ Frame* FrameTree::findBySpecifiedName(const AtomString& specifiedName, Frame& ac
+        if (!current)
+            return nullptr;
+
+        if (path.at(i) >= current->tree().childCount())
```
This seems redundant.  If this is the case, current will become null.

achristensen07 commented on Source/WebCore/page/FrameTree.cpp:
```diff
@@ -346,6 +346,22 @@ Frame* FrameTree::findBySpecifiedName(const AtomString& specifiedName, Frame& ac
+        if (path.at(i) >= current->tree().childCount())
+            return nullptr;
+
+        current = current->tree().child(path.at(i));
```
I think I prefer path[i] when accessing a vector, but at works, too.

achristensen07 commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -6380,7 +6380,21 @@ void WebPageProxy::findTextRangesForStringMatches(const String& string, OptionSe
 void WebPageProxy::replaceFoundTextRangeWithString(const WebFoundTextRange& range, const String& string)
 {
-    send(Messages::WebPage::ReplaceFoundTextRangeWithString(range, string));
+    getAllFrames([weakThis = WeakPtr { *this }, range, string](std::optional<FrameTreeNodeData>&& frames) mutable {
```
This seems like a wasted set of IPCs to me to gather data that is mostly not used.  Can't we just traverse the WebFrameProxy tree instead?  Ditto for all use of getAllFrames in this PR.

achristensen07 commented on Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
```diff
@@ -571,8 +571,8 @@ + (WKTextPosition *)textPositionWithRect:(CGRect)positionRect;
 
-@property (nonatomic, copy) NSString *frameIdentifier;
 @property (nonatomic) NSUInteger order;
+@property (nonatomic, copy) NSMutableArray *pathToFrame;
```
I think this doesn't need to be mutable, does it?

achristensen07 commented on Source/WebCore/page/FrameTree.cpp:
```diff
@@ -346,6 +346,22 @@ Frame* FrameTree::findBySpecifiedName(const AtomString& specifiedName, Frame& ac
     }, activeFrame);
 }
 
+Frame* FrameTree::findByPath(const Vector<size_t>& path) const
```
This is only ever called on the main frame, and it would be problematic if this were ever called on a frame that isn't the main frame.  I think this logic would be better located on WebCore::Page.

colelao committed:
Fix UITextSearching protocol functions in site isolation
https://bugs.webkit.org/show_bug.cgi?id=302986
rdar://165243794

Reviewed by Sihui Liu and Alex Christensen.

This patch fixes issues with the `UITextSearching` protocol
in site isolation.

It fixes `replaceFoundTextInRange`, `decorateFoundTextRange`,
`scrollRangeToVisible`, `clearAllDecoratedFoundText`, and
`requestRectForFoundTextRange` in site isolation.

The main problem is that a client can search for string in
one WebView and then highlight or replace or scroll in another
that is hosting the same content. We address this issue by storing
the path from the mainFrame to the frame in which the match was
found on each WebFoundTextRange.

This way, we can traverse the frame tree for each
WebFoundTextRange and find, positionally, which frame the range
was found in. This means we do not have to store the frameIdentifier,
which will be different between WebViews.

* Source/WebCore/page/Frame.cpp:
(WebCore::Frame::indexInFrameTreeSiblings const):
(WebCore::Frame::pathToFrame const):
* Source/WebCore/page/Frame.h:
* Source/WebCore/page/Page.cpp:
(WebCore::Page::findFrameByPath const):
* Source/WebCore/page/Page.h:
* Source/WebKit/Shared/WebFoundTextRange.cpp:
(WebKit::WebFoundTextRange::operator== const):
(WebKit::operator<<):
* Source/WebKit/Shared/WebFoundTextRange.h:
(WTF::HashTraits<WebKit::WebFoundTextRange>::constructDeletedValue):
(WTF::HashTraits<WebKit::WebFoundTextRange>::isDeletedValue):
* Source/WebKit/Shared/WebFoundTextRange.serialization.in:
* Source/WebKit/UIProcess/WebFrameProxy.cpp:
(WebKit::WebFrameProxy::childFrame const):
* Source/WebKit/UIProcess/WebFrameProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::replaceFoundTextRangeWithString):
(WebKit::WebPageProxy::scrollTextRangeToVisible):
(WebKit::WebPageProxy::clearAllDecoratedFoundText):
(WebKit::WebPageProxy::didBeginTextSearchOperation):
(WebKit::WebPageProxy::requestRectForFoundTextRange):
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(+[WKFoundTextRange foundTextRangeWithWebFoundTextRange:]):
(-[WKFoundTextRange dealloc]):
(-[WKFoundDOMTextRange webFoundTextRange]):
(-[WKFoundPDFTextRange webFoundTextRange]):
* Source/WebKit/WebProcess/WebPage/WebFoundTextRangeController.cpp:
(WebKit::WebFoundTextRangeController::findTextRangesForStringMatches):
(WebKit::WebFoundTextRangeController::rectsForTextMatchesInRect):
(WebKit::WebFoundTextRangeController::frameForFoundTextRange const):
* Tools/TestRunnerShared/spi/UIKitSPIForTesting.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPage.mm:
(TEST(WebKit, FindAndReplace)):
(swizzledIsEmbeddedScreen): Deleted.
(-[TestScrollViewDelegate init]): Deleted.
(-[TestScrollViewDelegate scrollViewDidEndScrollingAnimation:]): Deleted.
(-[TestFindDelegate setDidAddLayerForFindOverlayHandler:]): Deleted.
(-[TestFindDelegate didAddLayerForFindOverlayHandler]): Deleted.
(-[TestFindDelegate setDidRemoveLayerForFindOverlayHandler:]): Deleted.
(-[TestFindDelegate didRemoveLayerForFindOverlayHandler]): Deleted.
(-[TestFindDelegate _webView:didAddLayerForFindOverlay:]): Deleted.
(-[TestFindDelegate _webViewDidRemoveLayerForFindOverlay:]): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPageUtilities.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPageUtilities.mm:
(swizzledIsEmbeddedScreen):
(-[TestScrollViewDelegate init]):
(-[TestScrollViewDelegate scrollViewDidEndScrollingAnimation:]):
(-[TestFindDelegate setDidAddLayerForFindOverlayHandler:]):
(-[TestFindDelegate didAddLayerForFindOverlayHandler]):
(-[TestFindDelegate setDidRemoveLayerForFindOverlayHandler:]):
(-[TestFindDelegate didRemoveLayerForFindOverlayHandler]):
(-[TestFindDelegate _webView:didAddLayerForFindOverlay:]):
(-[TestFindDelegate _webViewDidRemoveLayerForFindOverlay:]):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm:
(TestWebKitAPI::(SiteIsolation, FindStringInFrameAndReplaceIOS)):
(TestWebKitAPI::(SiteIsolation, DecorateFoundTextRangeIOS)):
(TestWebKitAPI::(SiteIsolation, ScrollTextRangeToVisibleIOS)):
(TestWebKitAPI::(SiteIsolation, ClearAllDecoratedFoundTextIOS)):
(TestWebKitAPI::(SiteIsolation, RequestRectForFoundTextRangeIOS)):

Canonical link: https://commits.webkit.org/303851@main
